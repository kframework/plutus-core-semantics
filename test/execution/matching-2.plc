(program

(module Foo (import) (export () ())
  (define result
    (case (con Prelude.MkPair 1 2)
      (Prelude.MkPair (x y) (con Prelude.MkPair y x)))))

)
