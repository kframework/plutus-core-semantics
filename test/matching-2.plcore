(program

(module Foo (imported) (exported () ()) (local () ())
  (define result
    (case (con Nat.S (con Nat.Z))
          (Nat.S (x) (con Nat.Z))
          (Nat.Z ()  -1))))

)
