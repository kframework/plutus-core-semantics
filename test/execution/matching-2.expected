<T>
  <k> . </k>
  <lenv> .Map </lenv>
  <genv> ( Foo . result ) |-> ( con Prelude . MkPair 2  ( 1  .ValList ) ) </genv>
  <modules>
    <module>
      <curr> 0 </curr>
      <name> Foo </name>
      <data> .TyExpList </data>
      <decl> .NameList </decl>
      <import> .ModNmList </import>
      <exportTy> .TyExpList </exportTy>
      <exportTm> .NameList </exportTm>
    </module>
  </modules>
</T>
